### PR TITLE
implement table type access modifiers

### PR DESCRIPTION
**Pull request by jackdotink**
Implements table type access modifiers:
```luau
type foo = {
    read bar: number
    write baz: string,
}
```

### CODE REVIEW

File: full-moon/src/ast/luau.rs [hunk 1]
```diff
@@ -19,6 +19,8 @@ pub enum TypeInfo {
     Array {
         /// The braces (`{}`) containing the type info.
         braces: ContainedSpan,
+        /// The access modifer of the array, `read` in `{ read number }`.
+        access: Option<TokenReference>,
```
Comment (Kampfkarren) [inline full-moon/src/ast/luau.rs:23]: You don't have this in the `display`

Comment (Kampfkarren) [inline full-moon/src/ast/luau.rs:23]: Add `#[cfg_attr(feature = "serde", serde(skip_serializing_if = "Option::is_none"))]`

Comment (Kampfkarren) [reply full-moon/src/ast/luau.rs:29]: You didn't do this

**Comment by Kampfkarren** [comment]
Need tests for this as well

File: full-moon/src/ast/luau.rs (hunk 1 above)
Comment (jackdotink) [reply full-moon/src/ast/luau.rs:29]: I added `serde_option` or whatever it is to the entire enum. If you'd rather have this, I can do that instead.

**Review by DataM0del — COMMENTED** [review]
This should require the [syntax property access modifiers](https://github.com/luau-lang/rfcs/blob/master/docs/syntax-property-access-modifiers.md) RFC to be implemented into the luau-lang/luau repository.
I don't think it's implemented yet, and I tried the example provided in the PR description and ran it with Luau, but it gave me a syntax error.

**Comment by jackdotink** [comment]
@DataM0del try again because it works for me

**Comment by DataM0del** [comment]
> @DataM0del try again because it works for me

I'll update my Luau version. I wish though that Luau would have a --version command but the PR for that isn't moving because of complaints about how you have to change the version macro value to the current version or etc.

**Comment by DataM0del** [comment]
image
I updated. It works now.

**Comment by DataM0del** [comment]
> @DataM0del try again because it works for me

Actually, it doesn't work because of the trailing comma.

File: full-moon/src/ast/luau.rs (hunk 1 above)
Comment (Kampfkarren) [reply full-moon/src/ast/luau.rs:29]: Do you mean `display_option`? That's for display: the thing I gave is to not include it in serde if it's unset.

File: full-moon/src/ast/luau.rs [hunk 2]
```diff
@@ -266,8 +274,9 @@ pub enum IndexedTypeInfo {
 /// The `foo: number` in `{ foo: number }`.
 #[derive(Clone, Debug, Display, PartialEq, Node, Visit)]
 #[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
-#[display("{key}{colon}{value}")]
+#[display("{}{key}{colon}{value}", display_option(access))]
 pub struct TypeField {
+    pub(crate) access: Option<TokenReference>,
```
Comment (Kampfkarren) [inline full-moon/src/ast/luau.rs:279]: This also needs the serde thing

**Pull request merged**